src/DocumentFormat.OpenXml.Linq/OpenXmlLinqExtensions.cs

```diff
@@ -0,0 +1,169 @@
+        /// </summary>
+        /// <param name="part">The part to provide an <see cref="XElement"/> instance.</param>
+        /// <returns>A <see cref="IPartRootXElementFeature"/>.</returns>
+        public static IPartRootXElementFeature GetPartRootXElementFeature(this OpenXmlPart part)
+        {
+            var feature = part.Features.Get<IPartRootXElementFeature>();
+
+            if (feature is not null)
+            {
+                return feature;
+            }
+
+            part.TryAddDisposableFeature();
+            part.TryAddPartRootEventsFeature();
+
+            var xelement = new RootXElementFeature(part);
+
+            part.Features.SetDisposable<IPartRootXElementFeature>(xelement);
+
+            return xelement;
+        }
+
+        /// <summary>
+        /// Gets an <see cref="XElement"/> representation of the <paramref name="part"/>.
+        /// </summary>
+        /// <param name="part">The part to get the contents of.</param>
+        /// <returns>An <see cref="XElement"/>.</returns>
+        public static XElement? GetXElement(this OpenXmlPart part)
+            => part.GetPartRootXElementFeature().Root;
+
+        /// <summary>
+        /// Sets an <see cref="XElement"/> to the part. This will force a save to the underlying stream.
+        /// </summary>
+        /// <param name="part">The part to set to.</param>
+        /// <param name="xElement">The element to set.</param>
+        public static void SetXElement(this OpenXmlPart part, XElement? xElement)
+            => part.GetPartRootXElementFeature().Root = xElement;
+
+        /// <summary>
+        /// Saves the current <see cref="XElement"/> to the part.
+        /// </summary>
+        /// <param name="part">The part to save to.</param>
+        public static void SaveXElement(this OpenXmlPart part)
+            => part.GetPartRootXElementFeature().Save();
+
+        private sealed class RootXElementFeature : IPartRootXElementFeature, IDisposable
+        {
+            private readonly OpenXmlPart _part;
+            private readonly IPartRootEventsFeature _events;
+
+            private XElement? _rootXElement;
+
+            public RootXElementFeature(OpenXmlPart part)
+            {
+                _part = part;
+                _events = _part.Features.GetRequired<IPartRootEventsFeature>();
+
+                _events.Change += OnPartRootChange;
+            }
+
+            private void OnPartRootChange(FeatureEventArgs<OpenXmlPart> obj)
+            {
+                if (_part == obj.Argument)
+                {
+                    _rootXElement = null;
+                }
+            }
+
+            /// <summary>
+            /// Gets or sets the root LINQ to XML element.
+            /// </summary>
+            public XElement? Root
+            {
+                get
+                {
+                    if (_rootXElement is null)
+                    {
+                        if (_part.RootElement is null)
```

Comment:
```suggestion
                        if (!_part.IsRootElementLoaded)
```

@twsouthwick, see my longer comment below.

It would be better to invert the boolean expression and swap the `if` and `else` blocks below.